src/traces/bar/plot.js

```diff
@@ -102,12 +102,23 @@ function plot(gd, plotinfo, cdModule, traceLayer, opts) {
             var y0 = xy[1][0];
             var y1 = xy[1][1];
 
-            var isBlank = di.isBlank = !(
-                isNumeric(x0) && isNumeric(x1) &&
-                isNumeric(y0) && isNumeric(y1) &&
-                (x0 !== x1 || (shouldDisplayZeros && isHorizontal)) &&
-                (y0 !== y1 || (shouldDisplayZeros && !isHorizontal))
+            var isBlank = (
+                x0 === x1 ||
+                y0 === y1 ||
+                !isNumeric(x0) ||
+                !isNumeric(x1) ||
+                !isNumeric(y0) ||
+                !isNumeric(y1)
             );
+            // display zeros if line.width > 0
+            if(isBlank && shouldDisplayZeros) {
+                if(isHorizontal ? x1 - x0 === 0 : y1 - y0 === 0) {
+                    if(di.mlw || trace.marker.line.width > 0) {
```

Comment:
It's unclear whether this means `di.mlw || (trace.marker.line.width > 0)` or `(di.mlw || trace.marker.line.width) > 0`. Both are equivalent in this scenario, but to me 

```js
if(di.mlw > 0 || trace.marker.line.width > 0)
```

or simply

```js
if(di.mlw || trace.marker.line.width)
```

would be better.

---

Moreover, I would prefer if we combined all three `if()` into one. I don't see the need to have :three: if statements here. Oh well, this part in non-blocking. I'll let @archmoj decide what's best.

Reply:
@etpinard quick question:
I am wondering if `di.mlw` could be a negative number?

Reply:
Probably, as we don't coerce `arrayOk` attributes. That logic must be in a `Drawing.style*` method (please double-check though), so at this stage in `plot.js` we should **not** assume that `di.mlw >= 0`. Good eye!